app/models/goal.py

```diff
@@ -1,6 +1,37 @@
 from flask import current_app
 from app import db
+from app.models.task import Task
+
 
 
 class Goal(db.Model):
-    goal_id = db.Column(db.Integer, primary_key=True)
+    goal_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    # list holding all tasks associated with this goal
+    tasks = db.relationship('Task', back_populates='goal')
+
+    def to_dict(self):
+        if self.tasks == []:
+            return {
+                "id": self.goal_id,
+                "title": self.title,
+            }
+        else:
+            return {
+                "id": self.goal_id,
+                "task_ids": self.tasks
```

Comment:
This part of the function doesn't quite do what you are looking for. I noticed you handled this in your routes by creating a separate response body that takes the `task_ids` from the response body. This is a good choice. You could also refactor this method to get the appropriate task_ids 
```suggestion
            return {
                "id": self.goal_id,
                "task_ids": [task.task_id for task in self.tasks]
```

Reply:
As a full for loop

```
task_ids = []
for task in self.tasks:
    task_ids.append(task.id)